core/src/main/scala/cats/data/Chain.scala

```diff
@@ -320,17 +320,17 @@ sealed abstract class Chain[+A] {
   /**
    * Returns the number of elements in this structure
    */
-  final def length: Int = {
+  final def length: Long = {
     val iter = iterator
-    var i: Int = 0
+    var i: Long = 0
```

Comment:
just curious, did we actually tried one instance with more than 2 billion items?

Reply:
No I don't think so, I just wanted to be consistent with NonEmptyList and NonEmptyVector :)